ci(local-build): updates build and tests scripts inline with shipping

dholmes opened the pull request:
This includes updates to the Readme, use of prettier, updates to eslint config, etc. Also removes lint-staged and directs devs to use npm run all.

This is being done as a pre-step to merging in the final dependabot change.

dholmes commented on __tests__/inputs.test.ts:
```diff
@@ -1,34 +1,34 @@
-    process.env['INPUT_SOURCE'] = 'com.github.actions';
-    process.env['INPUT_RESOURCES'] = resourcesInput;
-    process.env['INPUT_DETAIL-TYPE'] = 'action-test';
+  ['', undefined],
```
@mloberg - if we are preferring the 4 space vs 2 space let me know and I'll look into why.

mloberg replied:
We weren't previously applying to the `__tests__` directory

dholmes replied:
Assuming ok to format __tests__ - let me know if I should exclude.

mloberg commented on .github/workflows/security.yml:
```diff
@@ -0,0 +1,15 @@
+name: Security
```
This is a public repo. We don't have access to run this

dholmes replied:
removed. Thanks! That would have bit me on merge.

mloberg commented on CONTRIBUTING.md:
```diff
@@ -31,7 +31,17 @@ AWS_SESSION_TOKEN=xxx
 
 When writing commits, we try to follow [Angular Commit Message Conventions](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-format).
+
+## Publishing
```
Publishing is a little different for this action. We should follow the [versioning guide](https://github.com/actions/toolkit/blob/master/docs/action-versioning.md), which is similar, but also a bit different

mloberg replied:
I guess the only difference is we should also tag a full semver in addition to updated the `v1` tag

dholmes replied:
Updated publishing section - please review.

mloberg commented on .prettierrc.js:
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  singleQuote: true,
+  trailingComma: 'all',
+  printWidth: 120,
```
nit: let's make this 80 for this project. I bumped it to 120 on the other one because we had some long lines and the formatting looked funny at 80

dholmes replied:
Updated.

dholmes committed:
chore(local-build): standardizes build and test scripts with other projects

This includes updates to the Readme, use of prettier, updates to eslint
config, etc. Also removes lint-staged and directs devs to use npm run
all.

This is being done as a pre-step to merging in the final dependabot
change.

mloberg commented on CONTRIBUTING.md:
```diff
@@ -31,7 +31,47 @@ AWS_SESSION_TOKEN=xxx
+Once you are ready to release the current state of the default branch, you
+will need to create a release for the version.
+
+To publish a new version of this action, update the _v1_ tag and then force
```
We should create the semver version first before creating the v1 tag

dholmes replied:
Ah, I misread. Updated.

mloberg commented on action.yml:
```diff
@@ -4,13 +4,13 @@ author: Articulate Global, LLC
   event-bus-name:
     required: true
-    description: The name or ARN of the event bus to recieve the event.
+    description: The name or ARN of the event bus to receive the event.
```
haha, you caught both of my common spelling mistakes

dholmes replied:
Well, vscode did. ;-) I just saw the squiggles.
